### PR TITLE
add standard behavior when you double click window

### PR DESCRIPTION
**Pull request by jhaynie**
add standard behavior when you double click window title to maximize / unmaximize the window

### CODE REVIEW

File: app/hyperterm.js [hunk 1]
```diff
@@ -398,6 +398,20 @@ export default class HyperTerm extends Component {
...
+    if (this.clicks++ >= 2) {
+        if (this.maximized) {
+            this.rpc.emit('unmaximize');
+        } else {
+            this.rpc.emit('maximize');
+        }
+        this.clicks = 0;
+        this.maximized = !this.maximized;
+    } else {
+        this.clickTimer = setTimeout(() => this.clicks = 0, 400);
```
Comment (rauchg) [inline app/hyperterm.js]: Please clear this on `componentWillUnmount`

**Comment by rauchg** [comment]
Thanks a lot for your contribution.

The double click rate is OS-specific and sometimes user-settings-specific. What stopped you from registering the `dblclick` event?

In addition, a further double-click should re-set the window to its prior size. (Terminal.app seems to do that, if you need an example).

**Comment by jhaynie** [comment]
click and dblclick events on the same element are impossible (by design): http://www.quirksmode.org/dom/events/click.html

the functionality works where each click with invert the other to match terminal / iterm type behavior.  so if you click it will maximize. another double click will unmaximize.

certainly once we have settings capability we can make the settings configurable.

**Comment by rauchg** [comment]
My bad, I didn't realize electron had `unmaximize` :) This is brilliant!

**Comment by daryl** [comment]
This doesn't seem to be working for me.

**Pull request merged**